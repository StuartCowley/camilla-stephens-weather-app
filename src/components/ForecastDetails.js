import "../styles/ForecastDetails.css";

function ForecastDetails(props) {
  const { forecast } = props;
  const formattedDate = new Date(forecast.date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__min-temp">
        Min temperature: {forecast.temperature.min} &deg;C
      </div>
      <div className="forecast-details__max-temp">
        Max temperature: {forecast.temperature.max} &deg;C
      </div>
      <div className="forecast-details__humidity">
        Humidity: {forecast.humidity} %
      </div>
      <div className="forecast-details__wind">
        Wind speed: {forecast.wind.speed} mph
      </div>
    </div>
  );
}

export default ForecastDetails;
