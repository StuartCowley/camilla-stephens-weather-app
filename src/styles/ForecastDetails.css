.forecast-details {
    align-self: center;
    text-align: center;
}

.forecast-details__date {
    font-size: 1.5em;
}