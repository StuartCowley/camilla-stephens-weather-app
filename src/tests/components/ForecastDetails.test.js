import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 11111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 15,
    wind: {
      speed: 10,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("12 °C")).toHaveClass("forecast-details__min-temp");
    expect(getByText("22 °C")).toHaveClass("forecast-details__max-temp");
    expect(getByText("15 %")).toHaveClass("forecast-details__humidity");
    expect(getByText("10 mph")).toHaveClass("forecast-details__wind");
  });
});
