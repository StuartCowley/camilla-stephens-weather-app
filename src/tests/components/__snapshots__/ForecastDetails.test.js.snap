// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__min-temp"
    >
      <strong>
        Min temperature:
      </strong>
       12 °C
    </div>
    <div
      class="forecast-details__max-temp"
    >
      <strong>
        Max temperature:
      </strong>
       22 °C
    </div>
    <div
      class="forecast-details__humidity"
    >
      <strong>
        Humidity:
      </strong>
       15 %
    </div>
    <div
      class="forecast-details__wind"
    >
      <strong>
        Wind speed:
      </strong>
       10 mph
    </div>
  </div>
</DocumentFragment>
`;
